compiler/vm.nim

```diff
@@ -677,6 +679,23 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
         regs[ra].intVal = s[idx].ord
       else:
         stackTrace(c, tos, pc, formatErrorIndexBound(idx, s.len-1))
+    of opcLdStrIdxAddr:
+      # a = addr(b[c]); similar to opcLdArrAddr
+      decodeBC(rkNode)
+      if regs[rc].intVal > high(int):
+        stackTrace(c, tos, pc, formatErrorIndexBound(regs[rc].intVal, high(int)))
+      let idx = regs[rc].intVal.int
+      let s = regs[rb].node.strVal.addr # or `byaddr`
+      if idx <% s[].len:
+         # `makePtrType` not accessible from vm.nim
+        let typ = newType(tyPtr, nextId c.idgen, c.module.owner)
```

Comment:
This is not required, values in the VM are untyped. ``node.typ`` can remain nil.

Reply:
it's needed otherwise `assert typ != nil` fails in `maybeHandlePtr`.
And that assert is needed, the type is propagated to `derefPtrToReg` and used there in `case typ.kind`.

Reply:
That indicates serious design problems with ``derefPtrToReg`` but we can improve this later.